cli/src/semgrep/metrics.py

```diff
@@ -157,12 +165,43 @@ def configure(
 
         self.metrics_state = metrics_state or MetricsState.AUTO
 
+    # TODO(cooper): It would really be best if EngineType included all of the
+    # information here, but I am a bit concerned about changing it, since it is
+    # currently an enum. Ideally it would be more like osemgrep's Engine_type.t,
+    # but that seems difficult to render here, and would seem to require
+    # threading much more information through that type. Since we only really
+    # care about the additional information being bundeled for metrics, we'll
+    # just take some additional parameters here. Currently this is just for
+    # secrets, but the same would apply for (supply chain)-related information.
```

Comment:
Yes, we do, but this class is used instead of the atd generated thing. I could try and migrate everything to use the metrics type instead but this has a couple issues:
- it makes business logic depend on the definition as it exists in the atd
- it makes more difficult to have methods on the relevant type
- it requires more changes to use the atd based class instead of the currently used one